lib/compress.js

```diff
@@ -2893,7 +2920,7 @@ merge(Compressor.prototype, {
                 });
             }
         }
-        return self;
+        return self.evaluate(compressor)[0];
```

Comment:
Could you please add a few tests for this AST_Sub property optimization when `unsafe` is `false`?

Ideally testing when the AST_Sub property can be reduced to:
* an integer
* a string literal with no spaces
* a string literal with a space
* special values like null, undefined, NaN or any other you may think of.

And another test where the AST_Sub prop cannot be reduced due to a side effect